test/bases.py

```diff
@@ -275,6 +275,39 @@ def assertBeatable(self, beatable: bool):
         """Asserts that the game can be beaten with the current state"""
         self.assertEqual(self.multiworld.can_beat_game(self.multiworld.state), beatable)
 
+    def assertAccessWith(self, spots: typing.List[str], item_names: typing.List[str],
+                         initial_items: typing.List[str] = None, resolution_hint: str = "Location"):
+        """
+        Assert that when the items are not collect, the spots are not reachable, but when
+        the items are collected, the spots are accessible. Also assert that no other new spots can be
+        accessed when the items are collected.
+        The `resolution_hint` parameter specifies which kind of spot to look for. It can take
+        "Location", "Entrance" or "Region" as values. Defaults to "Location".
+        """
+        if initial_items is None:
+            initial_items = []
+        assert resolution_hint in ("Location", "Region", "Entrance")
+        state = CollectionState(self.multiworld)
+        for item in initial_items:
+            state.collect(self.get_item_by_name(item))
+        reachable_spot_before = self.multiworld.reachable_spot(state, self.player, resolution_hint)
+        for spot in spots:
+            self.assertFalse(state.can_reach(spot, resolution_hint, self.player),
+                             f"{spot} is reachable without {item_names}")
+        items = self.get_items_by_name(item_names)
+        for item in items:
+            state.collect(item)
+        reachable_spot_after = self.multiworld.reachable_spot(state, self.player, resolution_hint)
+        for spot in reachable_spot_before:
```

Comment:
These should be sets instead of lists, since every element is guaranteed to be unique and order doesn't matter. Then you can just `reachable_spot_after = reachable_spots() - reachable_spot_before`